src/pgen/homologous.cpp

```diff
@@ -0,0 +1,137 @@
+// © 2021. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+
+#include <cmath>
+#include "geometry/geometry.hpp"
+#include "monopole_gr/monopole_gr.hpp"
+#include "pgen/pgen.hpp"
+#include "/Work/Phoebus/phoebus/Tests/GRsolver/reader.hpp"
+
+// Homologously collapsing star.
+
+namespace homologous {
+
+void ProblemGenerator(MeshBlock *pmb, ParameterInput *pin) {
+  
+  PARTHENON_REQUIRE(
+       typeid(PHOEBUS_GEOMETRY) == typeid(Geometry::MonopoleSph); 
+      "Problem \"homolohous\" requires \"MonopoleSph\" \" geometry!");
+
+  auto &rc = pmb->meshblock_data.Get();
+
+  PackIndexMap imap;
+  auto v = rc->PackVariables(
+      {fluid_prim::density, fluid_prim::velocity, fluid_prim::energy, fluid_prim::bfield,
+       fluid_prim::ye, fluid_prim::pressure, fluid_prim::temperature, fluid_prim::gamma1},
+      imap);
+
+  const int irho = imap[fluid_prim::density].first;
+  const int ivlo = imap[fluid_prim::velocity].first;
+  const int ivhi = imap[fluid_prim::velocity].second;
+  const int ieng = imap[fluid_prim::energy].first;
+  const int ib_lo = imap[fluid_prim::bfield].first;
+  const int ib_hi = imap[fluid_prim::bfield].second;
+  const int iye = imap[fluid_prim::ye].second;
+  const int iprs = imap[fluid_prim::pressure].first;
+  const int itmp = imap[fluid_prim::temperature].first;
+  const int igm1 = imap[fluid_prim::gamma1].first;
+
+
+  
+  const Real Cv = pin->GetOrAddReal("homologous", "Cv", 1.0);
+  const Real gamma = pin->GetOrAddReal("homologous", "gamma", 1.3334);
+  const bool spherical = pin->GetOrAddReal("homologous", "spherical_coords", true);
```

Comment:
If you're pulling htese parameters from the input deck, you need to add them to the input deck. Easier is probably pulling htem from the eos block of the input deck
```C++
  const Real Cv = pin->GetOrAddReal("eos", "Cv", 1.0);
  const Real gamma = pin->GetOrAddReal("eos", "gamma", 1.3334);
```

Reply:
Fixed